src/components/NoteCard/Note.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import { useNote } from "../../contexts/Note/NoteContext";
+import styles from "./Note.module.css";
+
+export default function NoteCard({ note }) {
+  const { title, notes, timestamp, color, label, priority } = note;
+  const [options, setOptions] = useState(false);
+  const {
+    noteState,
+    setNote,
+    noteDispatch,
+    deleteNote,
+    deleteArchivedNote,
+    archiveNote,
+    restoreArchivedNote,
+  } = useNote();
+
+  const editNote = (note) => {
+    setNote({ ...note, isEdited: (note.isEdited = true) });
+    setNote(note);
+  };
+
+  const deleteHandler = (note) => {
+    if (noteState.archivedNotes?.includes(note)) {
+      deleteArchivedNote(note);
+    } else {
+      deleteNote(note);
+    }
+  };
+
+  const optionsHandler = () => {
+    setOptions((prev) => !prev);
+  };
+
+  return (
+    <div
+      className={`${styles.note} ${
+        options ? styles.enlargeNote : styles.defaultNote
+      } border-radius box-shadow position-relative`}
+      style={{ backgroundColor: `${color}` }}
+    >
+      <div className={`${styles.note_content} flex-column-wrap flex-top-left`}>
+        <h3 className={styles.note_title}>{title}</h3>
+        <p
+          className={styles.note_body}
+          dangerouslySetInnerHTML={{
```

Comment:
Read about wrapping your dangerously set HTML with DOMPurify in order to prevent Injection attacks

Reply:
Oh, I wasn't aware of this. Will look into this and fix it. Thank you for the feedback Piyush.